riak_test/yz_solrq_test.erl

```diff
@@ -124,21 +164,326 @@ confirm_requeue_undelivered([Node|_] = Cluster, PBConn, BKey, Index) ->
     timer:sleep(?MELT_RESET_REFRESH + 1000), %% wait for fuse reset
     drain_solrqs(Node),
     verify_search_count(PBConn, Index, Count),
+    lager:info("confirm_requeue_undelivered ok"),
+    ok.
+
+confirm_no_contenttype_data(Cluster, PBConn, BKey, Index) ->
+    set_index(Cluster, Index, 1, 100, 100),
+    Count = 1,
+    Count = put_no_contenttype_objects(PBConn, BKey, Count),
+    yz_rt:commit(Cluster, Index),
+    verify_search_count(PBConn, Index, 1),
+    lager:info("confirm_no_contenttype_data ok"),
+    ok.
+
+confirm_purge_strategy(Cluster, PBConn) ->
+    confirm_purge_one_strategy(Cluster, PBConn,
+                               {?BUCKET5, ?INDEX5},
+                               {?BUCKET6, ?INDEX6}),
+    confirm_purge_idx_strategy(Cluster, PBConn,
+                               {?BUCKET7, ?INDEX7},
+                               {?BUCKET8, ?INDEX8}),
+    confirm_purge_all_strategy(Cluster, PBConn,
+                               {?BUCKET9, ?INDEX9},
+                               {?BUCKET10, ?INDEX10}),
+    confirm_purge_none_strategy(Cluster, PBConn,
+                                {?BUCKET11, ?INDEX11},
+                                {?BUCKET12, ?INDEX12}),
+    ok.
+
+confirm_purge_one_strategy(Cluster, PBConn, Bucket1Index1, Bucket2Index2) ->
+    PurgeResults = do_purge(Cluster, PBConn, Bucket1Index1, Bucket2Index2,
+                            ?PURGE_ONE),
+    check_one_purged(PurgeResults),
+    lager:info("confirm_purge_one_strategy ok"),
+    ok.
+
+check_one_purged({{Index1Written, Index1SearchResults},
+                  {Index2Written, Index2SearchResults}} = TestResults) ->
+    Condition =
+        equal(Index1Written, Index1SearchResults)
+        and first_purged(Index2Written, Index2SearchResults)
+        orelse first_purged(Index1Written, Index1SearchResults)
+        and equal(Index2Written, Index2SearchResults),
```

Comment:
Why the mixture of the short-circuiting `orelse` with the strict `and` (instead of `andalso`)?